nipoppy_cli/nipoppy/logger.py

```diff
@@ -56,3 +58,26 @@ def add_logfile(logger: logging.Logger, fpath_log: StrOrPathLike) -> None:
     logger.addHandler(file_handler)
     logger.info(f"Writing the log to {fpath_log}")
     return logger
+
+
+def capture_warnings(logger: logging.Logger) -> logging.Logger:
```

Comment:
If you are asking why we are capturing warnings at all, it's so that they seen by the logger and written wherever the log is written (otherwise it would appear in the console but not in log files, for example).

If you are asking why this is not in the same function as e.g. `get_logger`, it's because I couldn't get it to work otherwise 😅